### PR TITLE
http error handling for external services

### PR DESCRIPTION
**Pull request by julyto**
Added http error handling: vehicule_occupancies and free_floating.
Changed the declaration on single line for enviroment variables  in the Dockerfile

### CODE REVIEW

**Comment by azime** [comment]
Can you add some unit tests  ?

File: internal/vehicleoccupancies/vehicleoccupancies.go [hunk 1]
```diff
@@ -30,16 +30,21 @@ func RefreshVehicleOccupanciesLoop(context *VehicleOccupanciesContext,
...
-	time.Sleep(10 * time.Second)
-	for {
-		err := RefreshVehicleOccupancies(context, predictionURI, predictionToken, connectionTimeout, location)
-		if err != nil {
-			logrus.Error("Error while reloading VehicleOccupancy data: ", err)
-		} else {
-			logrus.Debug("vehicle_occupancies data updated")
+	if (context.courses == nil || context.stopPoints == nil) ||
+		(len(*context.courses) == 0 || len(*context.stopPoints) == 0) {
+		logrus.Error("VEHICULE_OCCUPANCIES: routine Vehicule_occupancies stopped, no stopPoints or courses loaded at start")
```
Comment (kadhikari) [inline internal/vehicleoccupancies/vehicleoccupancies.go]: A 'return' as above

File: internal/vehicleoccupancies/vehicleoccupancies.go [hunk 2]
```diff
@@ -30,16 +30,21 @@ func RefreshVehicleOccupanciesLoop(context *VehicleOccupanciesContext,
...
-	// Wait 10 seconds before reloading vehicleoccupacy informations
-	time.Sleep(10 * time.Second)
-	for {
-		err := RefreshVehicleOccupancies(context, predictionURI, predictionToken, connectionTimeout, location)
-		if err != nil {
-			logrus.Error("Error while reloading VehicleOccupancy data: ", err)
-		} else {
-			logrus.Debug("vehicle_occupancies data updated")
+	if (context.courses == nil || context.stopPoints == nil) ||
+		(len(*context.courses) == 0 || len(*context.stopPoints) == 0) {
```
Comment (kadhikari) [inline internal/vehicleoccupancies/vehicleoccupancies.go:34]: (len(*context.courses) == 0 || len(*context.stopPoints) == 0) could be replaced by
(len(context.GetCourses()) == 0 || len(context.GetStopPoints()) == 0). These two list are protected while writing in functions InitCourse and InitStopPoint

**Review by kadhikari — COMMENTED** [review]
It is easy to add at least one test per modification?

File: internal/vehicleoccupancies/vehicleoccupancies.go [hunk 3]
```diff
@@ -47,32 +54,42 @@ func RefreshVehicleOccupancies(context *VehicleOccupanciesContext, predict_url u
...
 		return
 	}
-	for {
-		err := LoadRoutesForAllLines(context, navitiaURI, navitiaToken, connectionTimeout, location)
-		if err != nil {
-			logrus.Error("Error while reloading RouteSchedule data: ", err)
+
+	if (context.courses == nil || context.stopPoints == nil) ||
+		(len(*context.courses) == 0 || len(*context.stopPoints) == 0) {
+		logrus.Error("VEHICLE_OCCUPANCIES: routine Route_schedule stopped, no stopPoints or courses loaded at start")
```
Comment (kadhikari) [inline internal/vehicleoccupancies/vehicleoccupancies.go:82]: Same as above

**Comment by julyto** [comment]
@azime 
I don't see how to do a test on an external API.

@kadhikari 
I added 4 tests:
two to test for null pointers if files are not found
two to test map sizes to 0 if the files have no data.

File: internal/vehicleoccupancies/vehicleoccupancies_test.go [hunk 4]
```diff
@@ -51,6 +65,55 @@ func TestNewStoppointWithMissingField(t *testing.T) {
...
+	courseFileName = "extraction_courses0.csv"
+
+	vehiculeOccupanciesContext := &VehicleOccupanciesContext{}
+
+	uri, err := url.Parse(fmt.Sprintf("file://%s/", fixtureDir))
+	require.Nil(err)
+	course, err := LoadCourses(*uri, defaultTimeout)
+	require.Nil(err)
+	vehiculeOccupanciesContext.InitCourse(course)
+	//fmt.Printf("erreur: %s\n", err)
```
Comment (kadhikari) [inline internal/vehicleoccupancies/vehicleoccupancies_test.go:112]: Minor: delete this line

File: internal/vehicleoccupancies/vehicleoccupancies.go (hunk 2 above)
Comment (julyto) [reply internal/vehicleoccupancies/vehicleoccupancies.go:34]: StopPoints and Courese are loaded before starting the routine. So the two lists are initialized and must not be null if data is found.

File: internal/vehicleoccupancies/vehicleoccupancies_test.go (hunk 4 above)
Comment (julyto) [reply internal/vehicleoccupancies/vehicleoccupancies_test.go:112]: Yes, I deleted

**Review by kadhikari — APPROVED** [review]

**Review by azime — APPROVED** [review]

**Pull request merged**